ksqldb-functional-tests/src/test/resources/query-validation-tests/group-by.json

```diff
@@ -511,6 +605,17 @@
         "message": "Duplicate value columns found in schema: `COUNT` BIGINT"
       }
     },
+    {
+      "name": "map used in non-aggregate function in select when group by uses subscript",
```

Comment:
one more test (may or may not already be there) - we should group by a map and select a value within the map as a column:
```sql
SELECT col1, col1['foo'] AS foo FROM .. GROUP BY col1;
```

Reply:
Grouping by a map throws the exception: 
```
Stacktrace was: io.confluent.ksql.SchemaNotSupportedException: Key format does not support key schema.
format: KAFKA
schema: Persistence{schema=STRUCT<COL1 MAP<VARCHAR, VARCHAR>> NOT NULL, unwrapped=false}
reason: The 'KAFKA' format does not support type 'MAP'
```